pkg/instancetype/controller/vm/controller_test.go

```diff
@@ -487,32 +488,29 @@ var _ = Describe("Instance type and Preference VirtualMachine Controller", func(
 				Name: instancetypeObj.Name,
 				Kind: "foobar",
 			}
-
-			_, err := instancetypeController.Sync(vm, vmi)
-			Expect(err).To(MatchError(ContainSubstring("got unexpected kind in InstancetypeMatcher")))
-			testutils.ExpectEvents(recorder, common.FailedCreateVirtualMachineReason)
+			syncVM, err := instancetypeController.Sync(vm, vmi)
+			Expect(err).To(MatchError(ContainSubstring("failure to find instance type or preference resources referenced by the VirtualMachine")))
```

Comment:
Put substring into a const? Also, you should be able to drop `ContainSubstring`. Same for occurences below.